cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -128,13 +128,19 @@ func (c *client) EnsureNamespace(namespaceName string) error {
 		return errors.Wrap(err, "error creating core clientset")
 	}
 
+	namespaces, err := clientset.CoreV1().Namespaces().List(metav1.ListOptions{})
+	for _, ns := range namespaces.Items {
+		if ns.Name == namespaceName {
+			return nil
+		}
+	}
 	namespace := apiv1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: namespaceName,
 		},
 	}
 	_, err = clientset.CoreV1().Namespaces().Create(&namespace)
-	if err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
I would leave in the check for existence. There is a slim chance that the Get call could return a 404 not found, then something else creates the namespace, and then this line tries to create it.